src/bin/main.ml

```diff
@@ -92,7 +92,43 @@ let set_signal_if_supported signal handler =
   with Invalid_argument _ ->
     ()
 
-let start urls sandbox git =
+module Date = struct
+  let pretty d =
+    let tm = Unix.localtime (Int64.to_float d) in
+    Printf.sprintf "%02d:%02d:%02d"
+      tm.Unix.tm_hour tm.Unix.tm_min tm.Unix.tm_sec
+end
+
+module HTTP = Irmin_http_server.Make(Cohttp_lwt_unix.Server)(Date)
+
+let serve_http = ()
+
+let http_server uri git =
+  let timeout = 3600 in
+  let uri = match Uri.host uri with
+    | None   -> Uri.with_host uri (Some "localhost")
+    | Some _ -> uri in
+  let port, uri = match Uri.port uri with
+    | None   -> 8080, Uri.with_port uri (Some 8080)
+    | Some p -> p, uri in
+  let mode = `TCP (`Port port) in
+  Logs.info (fun f -> f "daemon: %s" (Uri.to_string uri));
+  Printf.printf "Server starting on port %d.\n%!" port;
+  begin match git with
+    | None -> (* in-memory store *)
+      let module HTTP = HTTP(In_memory_store.Store) in
```

Comment:
The store handle is shared by all the HTTP requests to read and write stuff. Clearly not ideal. Also, we pick a default branch (in this case `master`) to display when accessing the top-level  paths (such as`/read/<path>` which is a short-cut to `/tree/master/read/<path>`.

Obviously that API has a lot of [issues](https://github.com/mirage/irmin/issues?q=is%3Aopen+is%3Aissue+label%3A%22REST+API%22) but @dgageot wanted to experiment a bit with it so I resurrected it.